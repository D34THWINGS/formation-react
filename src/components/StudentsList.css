.studentsList {
  width: 100%;
  padding: 0.5rem;
  margin: 0;
  background: #444;
  box-sizing: border-box;
  list-style: none;
  flex: 1;
  border-radius: 5px;
}

.studentsList__student {
}
