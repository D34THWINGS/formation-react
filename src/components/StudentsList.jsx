import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledStudentsList = styled.ul`
  width: 100%;
  padding: 0.5rem;
  margin: 0;
  background: #444;
  box-sizing: border-box;
  list-style: none;
  flex: 1;
  border-radius: 5px;
`;

const DeleteButton = styled.button`
  float: right;
  margin-left: 5px;
  border: none;
  border-radius: 0;
  background: #666;
  color: #fff;
  cursor: pointer;
  
  &:hover {
    opacity: 0.8;
  }
`;

const StudentsList = ({ students, onDeleteStudent }) => (
  <StyledStudentsList>
    {students.map(student => (
      <li key={student.id}>
        {student.lastName.toUpperCase()}
        {' '}
        {student.firstName}
        <DeleteButton onClick={() => onDeleteStudent(student.id)}>X</DeleteButton>
      </li>
    ))}
  </StyledStudentsList>
);

StudentsList.propTypes = {
  onDeleteStudent: PropTypes.func.isRequired,
  students: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    lastName: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
  })).isRequired,
};

export default StudentsList;
