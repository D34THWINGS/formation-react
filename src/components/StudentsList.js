import React from 'react';

import './StudentsList.css';

const StudentsList = ({ students }) => (
  <ul className="studentsList">
    {students.map(student => (
      <li key={student.id}>{student.lastName.toUpperCase()} {student.firstName}</li>
    ))}
  </ul>
);

export default StudentsList;
