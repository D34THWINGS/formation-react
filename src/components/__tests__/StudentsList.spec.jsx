import React from 'react';
import { shallow } from 'enzyme';

import StudentsList from '../StudentsList';

describe('StudentsList', () => {
  it('should render a list of students', () => {
    const students = [{ id: 1, firstName: 'foo', lastName: 'bar' }];
    const wrapper = shallow(<StudentsList students={students} />);
    expect(wrapper).toMatchSnapshot();
  });
});
