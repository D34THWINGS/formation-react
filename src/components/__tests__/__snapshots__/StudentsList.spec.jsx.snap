// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentsList should render a list of students 1`] = `
<ul
  className="studentsList"
>
  <li
    key="1"
  >
    BAR
     
    foo
  </li>
</ul>
`;
